.github/workflows/ci.yaml

```
@@ -32,6 +32,54 @@ jobs:
       - name: Run tests
         run: tox run -e unit
 
+  terraform-test:
+    name: Terraform - Lint and Simple Deployment
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: lint charm module
+        run: |
+          pushd ./terraform
+          terraform init
+          terraform fmt
+          terraform validate
+          pushd ./tests
+          terraform init
+          terraform fmt
+          terraform validate
+          popd
+          popd
```

Could this be split in two steps maybe?
Easier to know what broke I think.
You could do something like this in order to avoid also the pushd/popd
```suggestion
      - name: lint charm module
        working-directory: ./terraform
        run: |
          terraform init
          terraform fmt
          terraform validate
      - name: lint test charm module
         working-directory: ./terraform/tests
         run: |
          terraform init
          terraform fmt
          terraform validate
```